src/components/WelcomeBack/WalletList.vue

```diff
@@ -1,42 +1,121 @@
 <template>
-  <div class="wallet-list">
-    <p class="text">
-      Select which wallet do you want to unlock
+  <Card class="wallet-list">
+    <p class="text header">
+      Unlock your wallet
     </p>
-    <ul class="list">
-      <div class="wallets">
-        <li v-for="wallet in wallets" :key="wallet.id">
-          <RedirectionOption
-            type="forward"
-            :link="`/welcome-back/unlock/${wallet.id}`"
-            :name="wallet.name || wallet.id.trim()"
-          />
-        </li>
+    <div class="list">
+      <div class="row wallets">
+        <Select v-model="currentWallet" :options="walletOptions" />
       </div>
-      <li>
-        <RedirectionOption
-          data-test="create-wallet"
-          type="forward"
-          link="/ftu/welcome"
-          name="Create import or recover a wallet"
+      <div class="row" @keydown.enter.esc.prevent="unlockWallet">
+        <el-input
+          @keydown.enter.esc.prevent="unlockWallet"
+          class="big"
+          data-test="password"
+          placeholder="Please input password"
+          v-model="password"
+          show-password
         />
-      </li>
-    </ul>
-  </div>
+        <p data-test="error-alert" v-if="unlockWalletError" class="error">
+          Invalid password
+        </p>
+      </div>
+      <div class="row" @keydown.enter.esc.prevent="unlockWallet">
+        <el-button
+          class="big"
+          ref="submit"
+          data-test="unlock-wallet"
+          @click="unlockWallet"
+          type="primary"
+          :disabled="disableButton"
+        >
+          Unlock
+        </el-button>
+      </div>
+      <div class="row">
+        <el-button data-test="create-wallet" type="text" @click="nextStep">
+          Create import or recover a wallet ></el-button
+        >
+      </div>
+    </div>
+  </Card>
 </template>
 
 <script>
-import RedirectionOption from '@/components/RedirectionOption'
+import Select from '@/components/Select'
+import Card from '@/components/card/Card'
+import { mapState } from 'vuex'
 
 export default {
   name: 'WalletList',
   components: {
-    RedirectionOption,
+    Select,
+    Card,
+  },
+  data() {
+    return {
+      currentWallet: {},
+      password: '',
+    }
+  },
+  created() {
+    this.currentWallet = this.walletOptions[0]
+  },
+  methods: {
+    nextStep() {
+      this.$router.push('/ftu/welcome')
+    },
+    unlockWallet() {
+      this.$store.dispatch('unlockWallet', {
+        walletId: this.currentWallet.value,
+        password: this.password,
+      })
+      this.sent = true
+    },
+    updateView() {
+      this.$router.push('/wallet/transactions')
+    },
+  },
+  watch: {
+    sessionId: function(newValue) {
+      if (newValue) {
+        this.updateView()
+      }
+    },
+    password() {
+      if (this.unlockWalletError) {
+        this.$store.commit('clearError', { error: this.unlockWalletError.name })
```

Comment:
I'd use [mapMutations](https://vuex.vuejs.org/api/#mapgetters) from vuex to avoid call `this.$store.commit`,